indra/newview/llviewerregion.cpp

```diff
@@ -1162,8 +1162,14 @@ void LLViewerRegion::killCacheEntry(LLVOCacheEntry* entry, bool for_rendering)
             child = entry->getChild();
         }
     }
-    // Kill the assocaited overrides
+#if 0    // Nope: do not do this !  The object may re-rez later in the same
+         // session (e.g. for neighbour region objects, depending on draw
+         // distance, while moving or caming around) and its override may
+         // not be re-sent by the simulator, causing a loss in the override
+         // data !  HB
```

Comment:
I need to test this more to fully understand these cases, but the object cache probably needs a larger redesign to properly handle the neighbor region object cases.  the main correctness criteria after cache eviction should be that if the overrides get evicted from the cache then the object itself will also be evicted from the cache, and when the viewer sends the cache miss message to the simulator the full object update and overrides will be re-sent.

Reply:
This fix, which has been in place in my viewer for weeks, resulted from the direct observation and reliable reproduction of the problem by me. Since the bug I observed happened just after the inclusion of that line in my code base, I could easily identify it as the culprit (the benefit of weekly releases is that such bugs are quickly identified and easy to trace back to a culprit code change).

However, I did not try and diagnose what happened at the cache level, but could also see issues with this line active, after relogs and missing overrides.